app/views/layouts/application.html.erb

```diff
@@ -2,13 +2,88 @@
+
+  <%=stylesheet_link_tag "application"%>
+
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.2/jquery.min.js"></script>
+  <script src="http://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/js/bootstrap.min.js"></script>
 </head>
 <body>
+  <main class="container">
+    <nav class="navbar navbar-default">
+      <div class="container-fluid">
+        <div class="navbar-header"><%=link_to "Betsy", root_path, class:"navbar-brand"%></div>
+        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
+          <div class="col-lg-6">
+            <%= form_tag(search_product_path, method: "get", class: "navbar-form navbar-left") do %>
+            <div class="input-group">
+              <%= text_field_tag(:q, nil, class: 'form-control', placeholder: "Search a product") %>
+              <span class="input-group-btn">
+                <%= submit_tag("Search", class: "btn btn-default") %>
+                <% end %>
+              </span>
+            </div>
+          </div>
+          <ul class="nav navbar-nav navbar-right">
+            <li class="active"><%= link_to "All Products", products_path%></li>
+
+            <% if !current_user.nil? && current_user.id != nil %>
+            <ul class="nav navbar-nav">
+              <li class="dropdown">
+                <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">My Account <span class="caret"></span></a>
+                <ul class="dropdown-menu">
+                  <li>
+                    <%= link_to "My Account", user_path(current_user.id) %>
+                  </li>
+                  <li>
+                    <%=link_to "My Products", user_products_path(current_user.id)%>
+                  </li>
+                  <li>
+                    <%=link_to "My Order Fulfillments", user_fulfillment_index_path(current_user.id)%>
+                  </li>
+                  <li role="separator" class="divider"></li>
+                  <li>
+                  <%=link_to "Edit My Account", edit_user_path(current_user.id)%>
+                  </li>
+                </ul>
+              </li>
+            </ul>
+
+            <li><%=button_to "Log Out", logout_path, method: :delete, class: "btn btn-default navbar-btn"%></li>
+            <% else %>
+            <li class="active"><%= link_to "Sign Up", new_user_path%></li>
+            <li><%=button_to "Log In", login_path, method: :get, class: "btn btn-default navbar-btn" %></li>
+            <% end %>
+
+            <% if session[:order_id]%>
+              <li><%=link_to order_path(session[:order_id]), class:'cart_link' do %>
+                <span class="glyphicon glyphicon-shopping-cart"></span>
+              <% end %>
+              <li>
+            <% end %>
+          </ul>
+        </div>
+      </div>
+    </nav>
+
+    <br>
+
```

Comment:
It's oh so tempting to add <br>'s here, believe me!  Try creating the separation you are looking with your styling/CSS